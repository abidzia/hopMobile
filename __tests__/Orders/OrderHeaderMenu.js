import "react-native";
import React from "react";

// Note: test renderer must be required after react-native.
import renderer from "react-test-renderer";
import { MenuContext } from "react-native-popup-menu";

import OrderHeaderMenu from "../../src/components/Orders/OrderHeaderMenu";

it("renders correctly", () => {
  navigation = { state: { params: { mode: "pending" } } };
  const tree = renderer.create(
    <MenuContext>
      <OrderHeaderMenu navigation={navigation} />
    </MenuContext>
  );
  expect(tree).toMatchSnapshot();
});

it("renders correctly", () => {
  navigation = { state: { params: null } };
  const tree = renderer.create(
    <MenuContext>
      <OrderHeaderMenu navigation={navigation} />
    </MenuContext>
  );
  expect(tree).toMatchSnapshot();
});
