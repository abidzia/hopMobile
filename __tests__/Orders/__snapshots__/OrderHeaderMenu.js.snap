// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
        undefined,
      ]
    }
  >
    <View
      style={undefined}
    >
      <View
        collapsable={false}
        style={undefined}
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          menuName="menu-1"
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              undefined,
              undefined,
            ]
          }
          testID={undefined}
        >
          <Text
            accessibilityTraits="header"
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Array [
                Object {
                  "color": "rgba(0, 0, 0, .9)",
                  "fontSize": 17,
                  "fontWeight": "600",
                  "marginHorizontal": 16,
                  "textAlign": "center",
                },
                Object {
                  "color": "#f57e20",
                  "paddingVertical": 20,
                },
              ]
            }
          >
            Pending Orders
              ▾
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`renders correctly 2`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
        undefined,
      ]
    }
  >
    <View
      style={undefined}
    >
      <View
        collapsable={false}
        style={undefined}
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          menuName="menu-2"
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              undefined,
              undefined,
            ]
          }
          testID={undefined}
        >
          <Text
            accessibilityTraits="header"
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Array [
                Object {
                  "color": "rgba(0, 0, 0, .9)",
                  "fontSize": 17,
                  "fontWeight": "600",
                  "marginHorizontal": 16,
                  "textAlign": "center",
                },
                Object {
                  "color": "#f57e20",
                  "paddingVertical": 20,
                },
              ]
            }
          >
            All Orders
              ▾
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
